src/JoomlaBrowser.php

```diff
@@ -442,28 +442,48 @@ public function uninstallExtension($extensionName)
 		$this->debug('Extension successfully uninstalled');
 	}
 
-	/**
-	 * Function to Search For an Item in Joomla! Administrator Lists views
-	 *
-	 * @param   String  $name  Name of the Item which we need to Search
-	 *
-	 * @return void
-	 */
-	public function searchForItem($name = null)
-	{
-		$I = $this;
-		if($name)
-		{
-			$I->debug("Searching for $name");
-			$I->fillField(['id' => "filter_search"],$name);
-			$I->click(['xpath' => "//button[@type='submit' and @data-original-title='Search']"]);
-		}
-		else
-		{
-			$I->debug('clearing search filter');
-			$I->click(['xpath' => "//button[@type='button' and @data-original-title='Clear']"]);
-		}
-	}
+    /**
+     * Function to Search For an Item in Joomla! Administrator Lists views
+     *
+     * @param   String  $name     Name of the Item which we need to Search. If null it cleans the searchbox
+     * @param   array   $options  Optional Array containing the locators of the search buttons
```

Comment:
I some how feel this kind of usage will make it difficult for someone new, I understand in default scenario it will work just fine, but in case of a different template, developer will have to look into the code to find the exact name they need to set in the array, I think instead of taking an array why don't you break all of them into different parameters with default values set to Joomla's default? because in that case if they use any good IDE they will automatically understand what value is required to be set for which parameter. 
something like this
```
public function searchForItem($name = null, $searchFieldLocator, $searchSubmitLocator, $searchClearLocator)
```
What do you think?